src/Ocelot/Security/IPSecurity/IPSecurityPolicy.cs

```diff
@@ -18,7 +18,7 @@ public async Task<Response> Security(DownstreamRoute downstreamRoute, HttpContex
 
             if (securityOptions.IPBlockedList != null)
             {
-                if (securityOptions.IPBlockedList.Exists(f => f == clientIp.ToString()))
+                if (securityOptions.IPBlockedList.Contains(clientIp.ToString()))
```

Comment:
ClientIp could be null, it's not handled

Reply:
Fixed!

Reply:
The funny fact is that YARP project developers sometimes implement operators: only [3 types](https://github.com/search?q=repo%3Amicrosoft%2Freverse-proxy%20operator&type=code) in whole project ! 🤣 
But they are Microsoft senior engineers with 150K+ salaries, right @ggnaegi ? 😄

Reply:
Maybe in a next Feature PR ? 🤣

Reply:
Yes, next time... 😋